agent/consul/server.go

```diff
@@ -380,21 +380,20 @@ func NewServer(config *Config, flat Deps, publicGRPCServer *grpc.Server) (*Serve
 	recorder := middleware.NewRequestRecorder(serverLogger)
```

Comment:
Maybe inject the recorder to the server by adding it as a param to NewServer. This way we can avoid the race that @mkeeler is referring to.